lib/engines/hckinstall/hckinstall.rb

```diff
@@ -276,31 +274,52 @@ def run_second(client:)
     end
 
     def prepare_setup_scripts_config
-      kit_string = @platform['kit']
-      kit_type = kit_string[0..2]
-      kit_version = ''
-      kit_type == 'HCK' || kit_version = kit_string[3..]
+      kit_type, kit_version = parse_kit_info
 
       config = {
         kit_type:,
         hlk_kit_ver: kit_version
       }
 
-      unless @kit_info['download_url'].nil?
-        download_kit_installer(@kit_info['download_url'],
-                               "#{kit_type}#{kit_version}", @hck_setup_scripts_path)
+      @kit_path = find_kit(kit_type, kit_version)
+
+      log_kit_status(@kit_path, kit_type, kit_version)
+
+      if @kit_path.nil?
+        @kit_path = download_kit_installer(@kit_info['download_url'],
+                                           "#{kit_type}#{kit_version}", @hck_setup_scripts_path)
       end
 
+      @kit_is_iso = @kit_path.end_with?('.iso')
+
+      @logger.info("HLK installer #{kit_type}#{kit_version} was found at #{@kit_path}")
+
+      create_setup_scripts_config(@hck_setup_scripts_path, config)
+    end
+
+    def parse_kit_info
+      kit_string = @platform['kit']
+      kit_type = kit_string[0..2]
+      kit_version = kit_type == 'HCK' ? '' : kit_string[3..]
+      [kit_type, kit_version]
+    end
+
+    def find_kit(kit_type, kit_version)
       installers = [
         "#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}Setup.exe",
-        "#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}/#{kit_type}Setup.exe"
+        "#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}/#{kit_type}Setup.exe",
+        "#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}Setup.iso"
       ]
 
-      raise unless (file = installers.find { File.exist? _1 })
-
-      @logger.info("HLK installer #{file} was found")
+      installers.find { File.exist? _1 }
+    end
 
-      create_setup_scripts_config(@hck_setup_scripts_path, config)
+    def log_kit_status(kit_presense, kit_type, kit_version)
+      if kit_presense
+        @logger.info("HLK installer #{kit_type}#{kit_version} already exists")
+      elsif @kit_info['download_url'].nil? && kit_presense.nil?
```

Comment:
- Remove `kit_presense.nil?`; It's `elsif` so it is always true.
- Raising is not about logging so move out from this method.
- presence, not presense.